datafusion/physical-plan/src/test.rs

```diff
@@ -17,27 +17,347 @@
 
 //! Utilities for testing datafusion-physical-plan
 
+use std::any::Any;
 use std::collections::HashMap;
+use std::fmt;
+use std::fmt::{Debug, Formatter};
 use std::pin::Pin;
 use std::sync::Arc;
+use std::task::Context;
+
+use crate::common;
+use crate::execution_plan::{Boundedness, EmissionType};
+use crate::memory::MemoryStream;
+use crate::metrics::MetricsSet;
+use crate::stream::RecordBatchStreamAdapter;
+use crate::streaming::PartitionStream;
+use crate::ExecutionPlan;
+use crate::{DisplayAs, DisplayFormatType, PlanProperties};
 
 use arrow::array::{Array, ArrayRef, Int32Array, RecordBatch};
 use arrow_schema::{DataType, Field, Schema, SchemaRef};
+use datafusion_common::{
+    config::ConfigOptions, internal_err, project_schema, Result, Statistics,
+};
 use datafusion_execution::{SendableRecordBatchStream, TaskContext};
-use futures::{Future, FutureExt};
+use datafusion_physical_expr::{
+    equivalence::ProjectionMapping, expressions::Column, utils::collect_columns,
+    EquivalenceProperties, LexOrdering, Partitioning,
+};
 
-use crate::memory::MemorySourceConfig;
-use crate::source::DataSourceExec;
-use crate::stream::RecordBatchStreamAdapter;
-use crate::streaming::PartitionStream;
-use crate::ExecutionPlan;
+use futures::{Future, FutureExt};
 
 pub mod exec;
 
+#[derive(Clone, Debug)]
+pub struct MockMemorySourceConfig {
```

Comment:
Some suggestions:
1. Document that this implements an in memory `DataSource` for testing purposes (and maybe we can add a link to the real memory source) and that it duplicates much of the `MemorySourceConfig` to avoid a dependency between physical-plan and datasource
2. Rename it to something like `TestMemoryExec`  as this is not a config but rather something that `impl ExecutionPlan` and is used as a source of data in the tests